src/pages/resultsView/fusion/ResultViewFusionMapperStore.ts

```diff
@@ -0,0 +1,212 @@
+                return f;
+            });
+        }
+    }, []);
+
+    /**
+     * Apply color scale to the transcript
+     * @param {EnsemblTranscriptExt} d
+     * @param {boolean} isRefGene
+     * @param {EnsemblTranscriptExt[]} tempArr
+     */
+    private applyColor(d: EnsemblTranscriptExt, isRefGene: boolean, tempArr: EnsemblTranscriptExt[]) {
+
+        const attributeName: string = isRefGene ? 'transcriptId' : 'geneId';
+        const colorScale = isRefGene ? REF_GENE_COLORSCALE : FUS_GENE_COLORSCALE;
+
+        // Repeat color when length more than color scale array
+        const colorScaleIdx = tempArr.length % colorScale.length;
+        d.fillColor = colorScale[colorScaleIdx];
+
+        // collect in a temp array
+        if (tempArr.filter(t => t[attributeName] === d[attributeName]).length === 0) {
+            tempArr.push(d);
+        }
+    }
+
+    /**
+     * Returns ensembl transcripts
+     * @type {MobxPromiseUnionTypeWithDefault<Array<EnsemblTranscript>>}
+     */
+    readonly ensemblTranscripts = remoteData({
+        await: () => [
+            this.fusionData,
+        ],
+        invoke: async () => {
+            // Collect all transcript id's from the fusion data
+            let transcriptIds: any = [];
+            if (this.fusionData.result) {
+                this.fusionData.result.forEach((fusion: StructuralVariant) => {
+                    if (transcriptIds.indexOf(fusion.site1EnsemblTranscriptId) < 0) {
+                        transcriptIds.push(fusion.site1EnsemblTranscriptId);
+                    }
+                    if (transcriptIds.indexOf(fusion.site2EnsemblTranscriptId) < 0) {
+                        transcriptIds.push(fusion.site2EnsemblTranscriptId);
+                    }
+                });
+            }
+            // Fetch ensembl transcript data with transcript ids as request params
+            return await fetchEnsemblTranscriptsByEnsemblFilter({'transcriptIds': transcriptIds})
+                .then((result: EnsemblTranscriptExt[]) => {
+                    let fusionTranscripts: EnsemblTranscriptExt[] = [];
+                    let refTranscripts: EnsemblTranscriptExt[] = [];
+                    return result.map(t => {
+                        // Check if transcript is reference gene
+                        t.isReferenceGene = t.hugoSymbols.indexOf(this.gene.hugoGeneSymbol) >= 0;
+                        this.applyColor(t, t.isReferenceGene, t.isReferenceGene ? refTranscripts : fusionTranscripts);
+                        return t;
+                    });
+                });
+        }
+    }, []);
+
+    /**
+     * Fetch pfam domain data
+     * @type {MobxPromiseUnionTypeWithDefault<PfamDomain[]>}
+     */
+    readonly pfamDomainData = remoteData<PfamDomain[] | undefined>({
+        await: () => [
+            this.ensemblTranscripts
+        ],
+        invoke: async () => {
+            if (this.ensemblTranscripts.result) {
```

Comment:
you know ensemblTranscripts is ready because you've awaited it, so this if is not necessary